src/+datools/+statistical/+ensemble/RHF.m

```diff
@@ -1,158 +1,167 @@
-        
+
         function analysis(obj, R, y)
             inflation = obj.Inflation;
             tc = obj.Model.TimeSpan(1);
-            
-            % define length of gaussian tail
-            gtl = obj.Truncate;
-            discrete_pts = obj.DiscretePoint;
-            
+
+            % define length of gaussian tail (not needed now)
+            %gtl = obj.Truncate;
+            %discrete_pts = obj.DiscretePoint;
+
+            % get the current ensemble forecast
             xf = obj.Ensemble;
             ensN = obj.NumEnsemble;
-            
+
+            % calculate mean and variance of the forecast
             xfm = mean(xf, 2);
             xfv = var(xf, 0, 2);
-            
+
             xas = zeros(size(xf));
-            
+
             % Inflate the forecast
             Af = xf - repmat(xfm, 1, ensN);
-            Af = inflation*Af;
+            Af = inflation * Af;
             xf = repmat(xfm, 1, ensN) + Af;
-            
+
             % Observable
             Hxf = obj.Observation.observeWithoutError(tc, xf);
+            % mean of observable
             Hxfm = mean(Hxf, 2);
-            
-            ind1 = 1:ensN-1; ind2 = 2:ensN;
-            
-            % do for each state variable
-            for i = 1:size(xf,1)
-                [xfs, sortin] = sort(xf(i,:));
-                
-                % try inflating the background
-                xfs = mean(xfs) + inflation*(xfs - mean(xfs));
-                
-                prior_ht = zeros(1, ensN + 1);
+
+            % will be used for slicing
+            ind1 = 1:ensN - 1;
+            ind2 = 2:ensN;
+
+            % assimilate each state variable
+            for i = 1:size(xf, 1)
+                % sort the forecast ensemble
+                %xfs = sorted forecast ensemble
+                [xfs, sortin] = sort(xf(i, :));
+
+                priorht = zeros(1, ensN+1);
                 % find the height of uniform part
                 for ii = 2:ensN
-                    prior_ht(ii) = 1/ ((ensN + 1)*abs((xfs(ii) - xfs(ii-1))));
+                    priorht(ii) = 1 / ((ensN + 1) * abs((xfs(ii) - xfs(ii-1))));
                 end
-                
+
                 Var = xfv(i);
-                
+
                 if strcmp(obj.Tail, 'Gaussian') == 1
-                    muleft = xfs(1) - sqrt(2*Var)*erfinv(2/(ensN + 1) -1);
-                    muright = xfs(end) - sqrt(2*Var)*erfinv(1 - 2/(ensN + 1));
-                    mu = [muleft muright];
+                    % find the mean of the left gaussian tail
+                    muleft = xfs(1) - sqrt(2*Var) * erfinv(2/(ensN + 1)-1);
+                    % find the mean of the right gaussian tail
+                    muright = xfs(end) - sqrt(2*Var) * erfinv(1-2/(ensN + 1));
+                    mu = [muleft, muright];
                 else
-                    fprintf('Error! Gaussian tail not provided!\n');
+                    fprintf('Error! Only Gaussian Tail is supported as of now!\n');
```

Comment:
if this is an error, call an error...

`error('Only Gaussian Tail is supported as of now.');`

Reply:
Done